addons/escoria-core/game/core-scripts/esc/esc_compiler.gd

```diff
@@ -150,6 +153,47 @@ func _compile(lines: Array) -> Array:
 				)
 				event.statements = self._compile(event_lines)
 			returned.append(event)
+		elif _match_regex.search(line):
+			var match_expr = ESCMatch.new(line)
+			escoria.logger.trace("Line is a match")
+			var match_lines = []
+			while lines.size() > 0:
+				var next_line = lines.pop_front()
+				if _comment_regex.search(next_line) or \
+					_empty_regex.search(next_line):
+					continue
+
+				var next_line_indent = \
+						escoria.utils.get_re_group(
+							_indent_regex.search(next_line),
+							INDENT_REGEX_GROUP
+						).length()
+				if next_line_indent > indent:
+					match_lines.append(next_line)
+				else:
+					lines.push_front(next_line)
+					break
+
+			if match_lines.size() > 0:
+				escoria.logger.trace(
+					"Compiling the next %d lines into the match expr" % \
+						match_lines.size()
+				)
+				var pairs = match_expr.parse(match_lines)
+				escoria.logger.trace("%d arms found for match" % pairs.size())
```

Comment:
Can we rename "arm" to something like "condition_option" please? It's not very intuitive.

Reply:
Or "branch".